### PR TITLE
feat: recommended-marquee block changes EXLM-2447

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please provide the Jira Issue your PR is for.
Please also provide test paths following the template below.

Jira ID: https://jira.corp.adobe.com/browse/EXLM-2447

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live
- After: https://exlm-2447--exlm--adobe-experience-league.hlx.live

### CODE REVIEW

**Review by gauthamksn — APPROVED** [review]

File: blocks/recommendation-marquee/recommendation-marquee.js [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+const targetEventEmitter = getEmitter('loadTargetBlocks');
+const UEAuthorMode = window.hlx.aemRoot || window.location.href.includes('.html');
+const DEFAULT_NUM_CARDS = 5;
+let placeholders = {};
+try {
+  placeholders = await fetchLanguagePlaceholders();
+} catch (err) {
+  // eslint-disable-next-line no-console
+  console.error('Error fetching placeholders:', err);
+}
```
Comment (noormd08) [inline blocks/recommendation-marquee/recommendation-marquee.js:26]: Add this placeholders inside the decorate method.

File: blocks/recommendation-marquee/recommendation-marquee.js [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+          if (selectedItem) {
+            fetchDataAndRenderBlock(selectedItem);
+          }
+        },
+      });
+    });
+  }
+
+  targetEventEmitter.on('dataChange', async (data) => {
+    console.log('-------------------- targetEventEmitter::', data);
```
Comment (noormd08) [inline blocks/recommendation-marquee/recommendation-marquee.js:747]: Remove this console.

File: scripts/adobe-target/adobe-target.js [hunk 3]
```diff
@@ -123,10 +144,13 @@ class AdobeTargetClient {
     const targetData = await this.getTargetData();
 
     if (targetData.length) {
+      this.sanitizeTargetData(targetData);
       this.targetArray = targetData.map(({ meta }) => {
         const { scope, 'criteria.title': criteriaTitle } = meta;
         const indexMatch = scope.match(/-(\d+)$/);
-        const targetIndex = indexMatch ? indexMatch[1] - 1 : null;
+        const considerMarquee = scope === 'exl-hp-auth-recs';
+        // eslint-disable-next-line no-nested-ternary
```
Comment (noormd08) [inline scripts/adobe-target/adobe-target.js:152]: Try avoiding these lines and update the code base.

File: blocks/recommendation-marquee/recommendation-marquee.css [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+  background-color: var(--spectrum-blue-900);
+  color: var(--spectrum-gray-50);
+  padding: 12px 25px 14px 25px;
+  margin-right: 0;
+}
+
+.recommendation-marquee-result-link a {
+  font-size: var(--spectrum-font-size-200);
+  line-height: 20px;
+  font-weight: bold;
```
[the remark this answers is not in the document]
Comment (nitin-rachabathuni) [reply blocks/recommendation-marquee/recommendation-marquee.css]: done done

File: blocks/recommendation-marquee/recommendation-marquee.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply blocks/recommendation-marquee/recommendation-marquee.js]: done done

File: scripts/adobe-target/adobe-target.js (hunk 3 above)
Comment (nitin-rachabathuni) [reply scripts/adobe-target/adobe-target.js]: done done

File: blocks/recommendation-marquee/recommendation-marquee.js [hunk 5]
```diff
@@ -0,0 +1,760 @@
...
+  return cardIdsToExclude.length <= 1 ? query : `(${query})`;
+};
+
+async function findEmptyFilters(targetCriteriaId, profileInterests = []) {
+  const removeFilters = [];
+  const resp = await defaultAdobeTargetClient.getTargetData(targetCriteriaId);
+  if (resp?.data) {
+    const { data } = resp;
+    profileInterests.forEach((interest) => {
+      let found = false;
```
[the remark this answers is not in the document]
Comment (nitin-rachabathuni) [reply blocks/recommendation-marquee/recommendation-marquee.js]: done done

File: blocks/recommendation-marquee/recommendation-marquee.css [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+  .recommendation-marquee.block
+    .browse-cards-block-content
+    .card-wrapper:nth-child(1)
+    .browse-card
+    .browse-card-footer {
+    width: 50%;
+    height: 20%;
+    float: left;
+  }
```
[the remark this answers is not in the document]
Comment (nitin-rachabathuni) [reply blocks/recommendation-marquee/recommendation-marquee.css:261]: while committing and pushing, it is auto formatting Noor.

**Review by noormd08 — APPROVED** [review]

**Pull request merged**